Simulation.kt

```diff
@@ -36,7 +36,8 @@ abstract class Simulation<CLIENT: DBClient<*>, out CONTEXT: Context<*, *>>(
     protected abstract val name: String
 
     fun init() {
-        init(randomSource.nextSource())
+        val randomSource2 = randomSource.nextSource()
+        if (context.recreateDatabase) init(randomSource2)
```

Comment:
So when `recreateDatabase` is false, we generate a new random source but just discard it?

While this is probably harmless, it reads awkwardly. We should not do that.
```suggestion
        if (context.recreateDatabase) init(randomSource.nextSource())
```

Reply:
Just noticed the comment in the PR description about incrementing the master random source whether the DB is recreated or not.

If that's the behaviour we want, then fine - I'd say it has the potential to blindside the reader though! Could we add a comment in the code itself clarifying that yes, this is really what we want to do here (and why)?

Reply:
Yes, that is the best way to ensure the random selections remain deterministic. Resolved as discussed.

Reply:
The following change was tested (lines 39 and 40):
```
if (context.recreateDatabase) init(randomSource.nextSource())
else randomSource.nextSource()
```
however this did not produce deterministic results. The following unit test was run:
```
@Test
fun testSource() {
    val seed: Long = 0
    val randomSourceA = RandomSource(seed)
    val randomSourceB = RandomSource(seed)
    assert(randomSourceA.nextInt(1000000) == randomSourceB.nextInt(1000000))

    val randomSourceA2 = randomSourceA.nextSource()
    repeat(1000000) {
        randomSourceA2.nextBoolean()
    }

    if (false) {}
    else randomSourceB.nextSource()

    assert(randomSourceA.nextInt(1000000) == randomSourceB.nextInt(1000000))
}
```
which passed. It is unknown why these changes to lines 39 and 40 affect the determinism of the simulation. An issue has been opened [here](https://github.com/vaticle/typedb-simulation/issues/145).